src/Handlers/CrawlRequestFulfilled.php

```diff
@@ -0,0 +1,105 @@
+<?php
+
+namespace Spatie\Crawler\Handlers;
+
+use Spatie\Crawler\Crawler;
+use Spatie\Crawler\CrawlUrl;
+use Spatie\Crawler\LinkAdder;
+use Spatie\Robots\RobotsMeta;
+use Spatie\Robots\RobotsHeaders;
+use Spatie\Crawler\CrawlSubdomains;
+use Psr\Http\Message\StreamInterface;
+use Psr\Http\Message\ResponseInterface;
+
+class CrawlRequestFulfilled
+{
+    /** @var \Spatie\Crawler\Crawler */
+    protected $crawler;
+
+    /** @var \Spatie\Crawler\LinkAdder */
+    protected $linkAdder;
+
+    public function __construct(Crawler $crawler)
+    {
+        $this->crawler = $crawler;
+
+        $this->linkAdder = new LinkAdder($this->crawler);
+    }
+
+    public function __invoke(ResponseInterface $response, $index)
+    {
+        $crawlUrl = $this->crawler->getCrawlQueue()->getUrlById($index);
+
+        $body = $this->convertBodyToString($response->getBody(), $this->crawler->getMaximumResponseSize());
+
+        $robotsHeaders = RobotsHeaders::create($response->getHeaders());
```

Comment:
This function feels cluttered with `RobotsHeaders` & `RobotsMeta` implementation details. Maybe we should create a local `Robots` class (created from a `ResponseInterface`) to deal with these?

Reply:
Good idea, I'll do that.